### PR TITLE
Admin Import Tool, Documentation, and Scenario Coverage

### PR DESCRIPTION
**Pull request by orneryd**
#171

## Summary

This PR improves Neo4j-compatibility and operational correctness for `nornicdb-admin` import/export, and fixes a cosine fast-path index-selection bug.

### Admin import/export updates
- Add/verify official `nornicdb-admin` bulk import docs and operational scenarios.
- Preflight source files before writes for clearer fail-fast behavior.
- Preserve relationship `:ID` values during import/export (`relationships.csv` now includes `:ID` round-trip).
- Enforce duplicate relationship ID detection with deterministic `ExitDuplicateID` behavior.
- Improve CLI exit-code propagation by mapping typed admin-import errors instead of collapsing to generic exit code `1`.
- Export schema DDL for relationship fulltext/vector index forms.
- Preserve real node property `id` on Neo4j CSV export (only skipped when it exactly matches internal `node.ID`).

### Cypher/vector correctness updates
- Persist and carry vector index `entityType` metadata (`NODE` vs `RELATIONSHIP`) through schema paths.
- Make cosine fast-path index lookup entity-type aware to avoid wrong index selection when label/property overlap.
- Add regression tests for entity-type disambiguation and relationship-ID duplicate handling.

## Example Usage

Basic import:

```bash
nornicdb-admin database import full mydb \
  --nodes=Person=people.csv \
  --relationships=KNOWS=relationships.csv \
  --data-dir=./data \
  --schema=./schema.cypher \
  --report-file=./import-report.json
  ```
Multi-file source import:
```
nornicdb-admin database import full mydb \
  --nodes=Person=people-header.csv,people-part-2.csv \
  --relationships=WORKS_AT=works-at-header.csv,works-at-part-2.csv
```
Separate databases in the same storage directory:
```
nornicdb-admin database import full alpha --nodes=Person=alpha.csv --data-dir=./data
nornicdb-admin database import full beta --nodes=Person=beta.csv --data-dir=./data
```

Notes
`--from-path` schema auto-discovery now documented as: prefer `schema.nornic.json`, fallback to `schema.cypher` (unless `--schema` is provided).
Re-importing into the same namespace is rejected because the database is no longer empty.
Duplicate relationship IDs now fail with typed duplicate-ID semantics instead of ambiguous generic failures.

### CODE REVIEW

File: pkg/adminimport/importer.go [hunk 1]
```diff
@@ -0,0 +1,996 @@
...
+	rowInFile   int
+	headerRead  bool
+	currentPath string
+}
+
+func openCSVSource(files []string, opts Options) (*csvSourceReader, error) {
+	if opts.Quote != '"' {
+		return nil, unsupported("custom --quote is not supported by the Go CSV reader yet")
+	}
+	return &csvSourceReader{files: files, opts: opts}, nil
```
[the remark this answers is not in the document]
Comment (orneryd) [reply pkg/adminimport/importer.go:756]: openCSVSource fail-fast: fixed in pkg/adminimport/importer.go. openCSVSource now iterates every file path with os.Stat before returning the reader, so a missing later file fails before any rows are imported. That behavior is covered by pkg/adminimport/importer_test.go in TestImporterScenario_MissingLaterSourceFailsBeforeWrites.

File: pkg/adminimport/importer.go [hunk 2]
```diff
@@ -0,0 +1,996 @@
...
+	var idSpaces []string
+	for i, col := range cols {
+		if i >= len(record) {
+			continue
+		}
+		value := record[i]
+		if value == "" || (opts.IgnoreEmptyStrings && value == "") {
+			continue
+		}
+		switch col.Kind {
```
[the remark this answers is not in the document]
Comment (orneryd) [reply pkg/adminimport/importer.go:376]: Node empty-string preservation: fixed in pkg/adminimport/importer.go. In [nodeFromRecord](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html), empty values are no longer unconditionally skipped. IDs still error if empty, and string properties are preserved as "" unless [IgnoreEmptyStrings](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) is set. That is covered by pkg/adminimport/importer_test.go in [TestImporterPreservesEmptyStringPropertiesUnlessIgnored](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html).

File: pkg/adminimport/importer.go [hunk 3]
```diff
@@ -0,0 +1,996 @@
...
+		}
+		if col.Kind != kindProperty {
+			continue
+		}
+		parsed, err := parsePropertyValue(record[i], col, opts)
+		if err != nil {
+			return nil, false, csvErr("", line, i+1, err)
+		}
+		props[col.Name] = parsed
+	}
```
[the remark this answers is not in the document]
Comment (orneryd) [reply pkg/adminimport/importer.go:516]: Relationship empty-string preservation: fixed in pkg/adminimport/importer.go. [edgeFromRecord](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) now handles empty relationship property values the same way: preserve empty string for string fields unless [IgnoreEmptyStrings](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) is enabled. Same regression test in pkg/adminimport/importer_test.go covers both node and relationship cases.

File: pkg/adminimport/importer.go [hunk 4]
```diff
@@ -0,0 +1,996 @@
...
+		return nil, "", csvErr("", line, len(cols)+1, fmt.Errorf("extra column without header"))
+	}
+	var id storage.NodeID
+	var lookupKey string
+	if len(idCols) > 0 {
+		id = storage.NodeID(strings.Join(idValues, "|"))
+		lookupKey = importIDKey(idSpaces, idValues)
+	} else {
+		id = storage.NodeID(fmt.Sprintf("_anon_%d", line-1))
+	}
```
[the remark this answers is not in the document]
Comment (orneryd) [reply pkg/adminimport/importer.go:437]: Anonymous node ID collisions across multi-file sources: fixed in pkg/adminimport/importer.go. IDs no longer come from per-file [line](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html); they come from [state.anonymousNodeSeq](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html), which is importer-wide. That is covered by pkg/adminimport/importer_test.go in [TestImporterMultiFileAnonymousNodeIDsRemainUnique](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html).

File: pkg/adminimport/importer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (orneryd) [reply pkg/adminimport/importer.go:756]: [openCSVSource](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) now loops over every file and calls [os.Stat](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) before returning the reader, so the “later file missing causes partial writes” issue is already fixed.

File: pkg/adminimport/neo4j_csv.go [hunk 5]
```diff
@@ -0,0 +1,770 @@
...
+	}
+	return true
+}
+
+func writeCSV(path string, header []string, rows [][]string, delimiter rune) error {
+	file, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
```
[the remark this answers is not in the document]
Comment (orneryd) [reply pkg/adminimport/neo4j_csv.go:479]: onfirmed, already updated in the current diff to use restrictive 0o600 permissions for exported CSV files as well.

File: pkg/adminimport/importer.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (orneryd) [reply pkg/adminimport/importer.go:376]: In pkg/adminimport/importer.go, [nodeFromRecord](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) no longer does the old unconditional skip on empty values. The current behavior is:

For [kindID](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html): empty value is an error.
For [kindProperty](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html): if the value is empty:
it is preserved as "" when [IgnoreEmptyStrings](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) is false and the column type is a preservable string type
it is skipped when [IgnoreEmptyStrings](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) is true
it is also skipped for non-string-like property types such as arrays and vectors
For labels and embeddings: empty values are skipped as before

File: pkg/adminimport/importer.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (orneryd) [reply pkg/adminimport/importer.go:516]: For the relationship property case in pkg/adminimport/importer.go, [edgeFromRecord](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) no longer does the old if i >= len(record) || record[i] == "" { continue } short-circuit. It now:

reads [value := record[i]](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html)
if [value == ""](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html):
preserves "" for eligible string properties when [IgnoreEmptyStrings](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) is false
skips it when [IgnoreEmptyStrings](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) is true
otherwise parses the property normally
So [--ignore-empty-strings](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) now has an effect for relationship properties too. This is covered by [TestImporterPreservesEmptyStringPropertiesUnlessIgnored](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) in pkg/adminimport/importer_test.go.
